### ISSUE TITLE
lexedata validate some more

### ISSUE THREAD

**Issue by Anaphory** [Anaphory/lexedata#81]
In addition to `cldf validate`, there are some more assumptions we make, which a validate script should check:
 - [ ] The primary key of a table is its #id.
 - [ ] Every #reference (#parameterReference, #languageReference, #cognatesetReference) corresponds to a foreign key relationship
 - [ ] #segments, #segmentSlice and #alignment correspond to each other in the obvious relationship. #segmentSlice is a valid python slice (i.e. "t e s t" has indices 1:4) into segments.
 - [ ] Empty forms may exist, but only if there is no actual form for the concept and the language, and probably given some other constraints.
 - [ ] All files should be in NFC normalized unicode
 
There are probably a few others, let's collect them here.
Many of them are not just tests, but could even be fixed automatically.

**Comment by Anaphory** [comment]
I have suggested No.s 1 and 2 to CLDF: https://github.com/cldf/cldf/issues/107

**Comment by Anaphory** [comment]
From the discussion there: There are acceptable design reasons that give explicit foreign keys the primacy over the semantics of #parameterReference and all the other references.

We can glue explicit foreign keys into the metadata by using the [`auto_constraints` method](https://github.com/cldf/pycldf/blob/master/src/pycldf/dataset.py#L344-L399). But essentially, this means we should overhaul quite a few scripts. Essentially, all the scripts that read some table into a dictionary indexed by #id and then glue that to some other table based on #xxxReference need to be changed to instead do the gluing based on the foreign key relationship.

@PaprikaSteiger how should we do this? Step by step as we go through the programs, or in one concentrated effort?

**Comment by Anaphory** [comment]
1. @xrotwang made the point that people might want ‘natural’ primary keys for some tables and all the machinery still works with #id being a normal column, so that one is not necessary.
2. The translation from references to foreign keys is done by pycldf at read-time, and will be strongly suggested in the new CLDF version.
3. I have started implementing the test for the CognateTable, and I'll ask @xrotwang for a review of that code, because I vaguely think that might also be interesting in core pycldf.
4. Handling of empty forms is still missing, but still open subject to #73.
5. Missing NFC-normalization of tables is not reported, but the normalization can be automatically done by runnning `lexedata.util` – to be fair, that's an obscure place to have that functionality.

In summary, there are some things left to do, but those are better addressed as separate issues.
